utils/tiles.py

```diff
@@ -74,3 +78,210 @@ def generate_tiles_coordinates(self, zooms):
                             self.filter_zooms_in_range(zooms), True):
             yield BoundingBoxTileCoordinatesGenerator.\
                 convert_mercantile(m_tile)
+
+
+class S3TileVerifier(object):
+    """
+    Used the logic of tilequeue to expand tile coordinates to the rawr and meta
+    tiles S3 locations that tilequeue would write to.
+
+    It is useful for arbitrary tile build verification because we can use the
+    methods provided here to pre-generate the s3 objects' paths that we need
+    to verify later.
+    """
+
+    def __init__(self,
+                 date_prefix,
+                 key_format_type_str,
+                 rawr_s3_bucket,
+                 meta_s3_bucket,
+                 rawr_tile_filename,
+                 high_zoom_tile_filename,
+                 low_zoom_tile_filename):
+        # type: (str, str, str, str, str, str, str) -> None
+        """
+        :param date_prefix: the build prefix string
+        :param key_format_type_str: the style of S3 key prefix
+        :param rawr_s3_bucket: the s3 bucket for storing rawr tiles
+        :param meta_s3_bucket: the s3 bucket for storing meta tiles
+        :param rawr_tile_filename: the local file name to store rawr tiles coords and its last updated time on S3
+        :param high_zoom_tile_filename: the local file name to store high_zoom meta tiles coords and its last updated time on S3
+        :param low_zoom_tile_filename: the local file name to store low_zoom meta tiles coords and its last updated time on S3
+        """
+        self.date_prefix = date_prefix
+        self.key_format_type_str = key_format_type_str
+        self.rawr_tile_filename = rawr_tile_filename
+        self.high_zoom_tile_filename = high_zoom_tile_filename
+        self.low_zoom_tile_filename = low_zoom_tile_filename
+        self.s3 = boto3.client('s3')
+        self.rawr_s3_bucket = rawr_s3_bucket
+        self.meta_s3_bucket = meta_s3_bucket
+
+    # https://github.com/tilezen/tilequeue/blob/9644d916a864bd7d97448c40823158016e5e6dd2/tilequeue/command.py#L1861
```

Comment:
probably want to explain that this expansion logic mimics this code at this URL

Reply:
fixed in the next commit